ts/client/src/utils/rpc.ts

```diff
@@ -171,20 +172,47 @@ const confirmTransaction = async (
     latestBlockhash.blockhash != null &&
     latestBlockhash.lastValidBlockHeight != null
   ) {
-    status = await connection.confirmTransaction(
-      {
-        signature: signature,
-        blockhash: latestBlockhash.blockhash,
-        lastValidBlockHeight: latestBlockhash.lastValidBlockHeight,
-      },
-      txConfirmationCommitment,
-    );
+    status = await Promise.race([
+      connection.confirmTransaction(
+        {
+          signature: signature,
+          blockhash: latestBlockhash.blockhash,
+          lastValidBlockHeight: latestBlockhash.lastValidBlockHeight,
+        },
+        txConfirmationCommitment,
+      ),
+      awaitTransactionSignatureConfirmation({
```

Comment:
this should also check the backup connections

Reply:
https://github.com/blockworks-foundation/mango-v4/blob/53bc52efe7c0b3eee01d4737be1f043c43cf9980/ts/client/src/utils/rpc.ts#L139